### ISSUE TITLE
[HOLD for payment 2023-06-02] [$1000] The transaction thread shows option to delete even for the payer

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19340]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
As UserA request money from user B
As UserB go to the IOU report
Click the IOUPreview to open the transaction details
In the transaction thread, click the three dots
## Expected Result:
There should be no delete button for a payer
## Actual Result:
It shows delete but only the requestor should be able to delete
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.16.5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
<img width="960" alt="Untitled" src="https://github.com/Expensify/App/assets/43996225/d112d617-c256-483a-90c7-35c18eb38867">
![image](https://github.com/Expensify/App/assets/43996225/9d0a97cd-8873-4466-898f-d00095e4b934)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @mountiny 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684333083150029

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bed0fba921c179b3</li>
        <li>Upwork Job ID: 1660491796198117376</li>
        <li>Last Price Increase: 2023-05-22</li>
    </ul>
</details>

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
transaction thread shows three dots options for payer too

### What is the root cause of that problem?
Currently, we are not using any condition to hide the threeDotMenu for the payer. 

### What changes do you think we should make in order to solve the problem?
We need to use the bool variable `isPayer` to include in the condition of `shouldShowThreeDotsButton`. Condition needs to be `!isPayer &&`
https://github.com/Expensify/App/blob/737862677811acc2b05c68209a56ecd58bee59f3/src/components/MoneyRequestHeader.js#L93

To correct the conditions for `isPayer` we need to update the instances of `props.report` to `moneyRequestReport` below.
https://github.com/Expensify/App/blob/737862677811acc2b05c68209a56ecd58bee59f3/src/components/MoneyRequestHeader.js#L87

**Comment by kadiealexander** [comment]
Reproed in chrome:

![image](https://github.com/Expensify/App/assets/59587260/c8cc53ea-1e4d-4f6f-9f38-66f44a915479)

**Comment by kamaljitSharma21** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Transaction thread shows three dots options for payer too

### What is the root cause of that problem?

We need to show ThreeDotMenu only to payee and not to the payer so for that we haven't added any condition as of now

### What changes do you think we should make in order to solve the problem?

we need to add a boolean variable isPayee in MoneyRequestHeader component

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/components/MoneyRequestHeader.js#L82

Here is the boolean variable - 
```
const isPayee = lodashGet(props.session, 'email', null) === lodashGet(props.parentReportAction, 'actorEmail', null);
```

Here need to update the condition of shouldShowThreeDotsButton prop for HeaderWithCloseButton by adding isPayee boolean into it this will solve our problem.

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/components/MoneyRequestHeader.js#L93

Here is the updated condition - 

```
shouldShowThreeDotsButton={!isSettled && props.isSingleTransactionView && isPayee}
```

**Comment by Santhosh-Sellavel** [comment]
@arosiclair I don't see the three dots option for anyone on the main, may this is no longer needed. Can you check this internally?

**Comment by mountiny** [comment]
@Santhosh-Sellavel If so that should be fixed too, there should be delete option for the one who has requested.

**Comment by Santhosh-Sellavel** [comment]
https://github.com/Expensify/App/assets/85645967/a4725237-fdaa-4fd1-b07a-58c1c805b5e4

Looking back, I don't see the three dots option on a chat report. I don't know what to call it as there is total Outstanding between the users

There is a chat for an IOU Transaction(Send/Request) but for this specific chat I see delete option for both users

cc: @mountiny @arosiclair

**Comment by Santhosh-Sellavel** [comment]
Sorry for the mix-up here. The exact issue said here is reproducible. And the first @Pujan92's proposal looks good to me! Let me know your thoughts thanks!

C+ Reviewed
👀 🎀 👀 

cc: @mountiny  @arosiclair

**Comment by mountiny** [comment]
I will take this one over since its related to the Manual reqeusts. I hope it works!

**Comment by mountiny** [comment]
Lets get the PR going @Pujan92, thanks!

**Comment by Pujan92** [comment]
@Santhosh-Sellavel @mountiny [PR](https://github.com/Expensify/App/pull/19433) is ready for review!

**Comment by kadiealexander** [comment]
Contracts sent to @Pujan92 and @Santhosh-Sellavel. Don't forget about the checklist above!

**Comment by kadiealexander** [comment]
Paid @Pujan92 for fix + speed bonus. Thanks for the speedy and awesome work Pujan!

**Comment by Santhosh-Sellavel** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/18148
> * [x]  [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/18148#issuecomment-1575124726
> * [x]  [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
> * [x]  [@Santhosh-Sellavel] Determine if we should create a regression test for this bug.
> * [x]  [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
> * [ ]  [@kadiealexander] Link the GH issue for creating/updating the regression test once the above steps have been agreed upon:

Looking back at the issue this seems to be missed handling on https://github.com/Expensify/App/pull/18148. Focus was just to get started with the new flow. No further discussion is needed here for any checklist updates.

**Comment by Santhosh-Sellavel** [comment]
### Regression Steps

1. Request money from user A to user B
2. Login as user B and open chat of user A
3. Click on IOU preview to open IOU Report
4. Verify the ThreeDotsMenu is not present in the header for the user because the user is the payer

👍  or 👎

**Comment by mountiny** [comment]
I think that works but also it should be something easily handled by the QA team in their normal regression testing. Happy to add it those, I will defer to @kadiealexander on this one

**Comment by kadiealexander** [comment]
I think these should be able to be followed. I'll raise the GH issue and if the QA team has any concerns I'll bring them back for discussion.

**Comment by kadiealexander** [comment]
And everyone is paid! Thanks for your hard work team.
